query/src/main/scala/filodb/query/exec/rangefn/RateFunctions.scala

```diff
@@ -329,8 +340,14 @@ class RateOverDeltaChunkedFunctionD extends ChunkedDoubleRangeFunction {
                                 endRowNum: Int): Unit =
     sumFunc.addTimeDoubleChunks(doubleVectAcc, doubleVect, doubleReader, startRowNum, endRowNum)
 
-  override def apply(windowStart: Long, windowEnd: Long, sampleToEmit: TransientRow): Unit =
-    sampleToEmit.setValues(windowEnd, sumFunc.sum / (windowEnd - (windowStart)) * 1000)
+  override def apply(windowStart: Long, windowEnd: Long, sampleToEmit: TransientRow): Unit = {
+    val curWindowStart = if (FiloQueryConfig.isInclusiveRange)
+                            windowStart
+                         else
+                            windowStart - 1
```

Comment:
Shouldn't we make the window smaller if the range is not inclusive?

Reply:
fix is to revert the new start-inclusive behavior if config is set to false